action.yml

```diff
@@ -7,36 +7,57 @@ inputs:
     default: 1.10.2
 runs:
   using: "composite"
+
   steps:
     - name: Set up cache paths
       id: cache-paths
       shell: bash
       run: |
-        echo "setupsbt_path=$HOME/work/_setupsbt" >> $GITHUB_OUTPUT
+        echo "sbt_toolpath=$RUNNER_TOOL_CACHE/sbt/${{ inputs.sbt-runner-version }}" >> "$GITHUB_OUTPUT"
+        echo "sbt_downloadpath=$RUNNER_TEMP/_sbt" >> "$GITHUB_OUTPUT"
+
+    - name: Check Tool Cache
+      id: cache-tool-dir
+      shell: bash
+      run: |
+        mkdir -p "${{ steps.cache-paths.outputs.sbt_downloadpath }}"
+        if [ -f "${{ steps.cache-paths.outputs.sbt_toolpath }}/sbt/bin/sbt" ]; then
+          echo "cache-hit=true" >> "$GITHUB_OUTPUT"
+        else
+          echo "cache-hit=false" >> "$GITHUB_OUTPUT"
+        fi
 
     - name: Cache sbt distribution
       id: cache-dir
+      if: steps.cache-tool-dir.outputs.cache-hit != 'true'
       uses: actions/cache@v4
       with:
-        path: ${{ steps.cache-paths.outputs.setupsbt_path }}
-        key: ${{ runner.os }}-sbt-${{ inputs.sbt-runner-version }}-1.1.1
+        path: ${{ steps.cache-paths.outputs.sbt_downloadpath }}
+        key: ${{ runner.os }}-sbt-${{ inputs.sbt-runner-version }}-1.1.3
+
+    - name: "Download sbt"
+      shell: bash
+      env:
+        SBT_RUNNER_VERSION: ${{ inputs.sbt-runner-version }}
+      if: steps.cache-tool-dir.outputs.cache-hit != 'true' && steps.cache-dir.outputs.cache-hit != 'true'
```

Comment:
I wonder if this should be `||` instead. I think we should download whenever we invalidate the setup-sbt version.

Reply:
So, initially I meant to have 2 different cache phase.
 1. Is it in the toolpath?  if yes skip steps 2, 3, and 4.
 2. Is it in the cache?  if yes skip step 3.
 3. Download to cache dir
 4. Unzip to toolpath.
 5. Add to `$PATH`

But I think that might have been confusing... so now its more like the following:
1. Is it in the toolpath?  if yes skip steps 2 and 3
2. Is it in the cache?  if yes skip step 3
3. Download, and Unzip to the toolpath
4. Add to `$PATH`

So this means for the `Download and Install sbt" we do want the condition to be `! ( isInToolPath == true || isInCache == true)` but since i can't write it like that, i flip it to `isInToolPath != true && isInCache != true`